### PR TITLE
new signing pipeline (part1)

### PR DESCRIPTION
**Pull request by TimothyMothra**
#2475.

This PR introduces the new pipelines for our new signed builds.
I need this on the `main` branch to test some of the compliance settings.

Please review screenshots below for a description of the pipeline output.

A follow up PR will:
- refactor/normalize some of the scripts and settings.
- resolve all compliance issues.

## Changes
- Creating three new pipelines
  - "Buddy" will be used in PR validation.
  - "Official" is for signing the AI SDK
  - "Official Redfield" is for signing the AI SDK w/ Redfield build flag
- Creating new *.cmd scripts.
  FYI: pipelines cannot have inline scripts and must use external files. 
  - i've numbered these scripts in order to sort them according to which "build stage" they will be used.

### Checklist
- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Comment by TimothyMothra** [comment]
As of commit `c6cfebc` we have three builds....

## Buddy
The buddy build does not permit signing and will eventually run with all PRs.

The log correctly identifies itself
image

The output NUPKGs are not signed.
The contained DLLs are not signed.
image

### TODO
- change build params to include build number. This is to distinguish between "official" builds.


## Official
The official build uses signing.

The log correctly identifies itself
image

The output NUPKGs are signed.
The contained DLLS are signed.
image



## Official Redfield
This is an official build, which uses signing.

The log correctly identifies itself
image

The output NUPKGs are signed.
The contained DLLs are signed.
image

### TODO
- should the package name include "redfield" instead of "beta"?

**Comment by TimothyMothra** [comment]
Fixed Redfield, now has correct package name.
image

**Comment by TimothyMothra** [comment]
fixed package name for Buddy build:
image

File: .pipelines/pipeline.user.windows.official.redfield.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    - !!defaultcommand
+      name: 'Install dependencies'
+      command: '.pipelines\01_restore.cmd'
+      fail_on_stderr: false
+
+build:
+  commands:
+    - !!buildcommand
+      name: 'Dotnet Build'
+      command: '.pipelines\02_build.cmd'
```
Comment (reyang) [inline .pipelines/pipeline.user.windows.official.redfield.yml:40]: nit: many of these commands (e.g. dotnet) are platform independent, consider getting rid of the cmd file.

e.g. https://github.com/open-telemetry/opentelemetry-dotnet/blob/66d2621170c614313c81fc82e4f6c79218244974/.github/workflows/linux-ci.yml#L44

File: .pipelines/pipeline.user.windows.official.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  commands:
+    - !!buildcommand
+      name: 'Dotnet Build'
+      command: '.pipelines\02_build.cmd'
+      arguments: '-p:PublicRelease=True -p:StableRelease=True -p:OfficialRelease=True'
+      fail_on_stderr: false
+      artifacts:
+        - from: '.\bin\Release\src'
+          include:
+            - '**/*.dll'
```
Comment (reyang) [inline .pipelines/pipeline.user.windows.official.yml:46]: it seems `\` and `/` are mixed up, consider using `/`.

**Review by reyang — APPROVED** [review]
LGTM with some suggestions.

File: .pipelines/pipeline.user.windows.official.redfield.yml (hunk 1 above)
Comment (TimothyMothra) [reply .pipelines/pipeline.user.windows.official.redfield.yml:40]: I would have GREATLY preferred this approach.
I went to office hours to ask this exact question. OneBranch does not permit in-line scripts as a rule!

In all of our github builds, i'm doing everything in-line:
https://github.com/microsoft/ApplicationInsights-dotnet/blob/e8eb5204d2757f7443c43cb546d6e3f03f201ae3/.github/workflows/build-and-test-BASE.yml#L47

File: .pipelines/pipeline.user.windows.official.yml (hunk 2 above)
Comment (TimothyMothra) [reply .pipelines/pipeline.user.windows.official.yml:46]: done

**Pull request merged**